mathesar/models/base.py

```diff
@@ -283,6 +283,60 @@ class Explorations(BaseModel):
     description = models.CharField(null=True)
 
 
+class Form(BaseModel):
+    id = models.UUIDField(primary_key=True)
+    name = models.CharField()
+    version = models.IntegerField()
+    database = models.ForeignKey('Database', on_delete=models.CASCADE)
+    base_table_oid = models.PositiveBigIntegerField()
+    schema_oid = models.PositiveBigIntegerField()
+    description = models.CharField(null=True)
+    header = models.OneToOneField('FormHeader', on_delete=models.DO_NOTHING)
+    submission = models.ForeignKey('SubmissionSettings', on_delete=models.CASCADE)
+    is_public = models.BooleanField(default=False)
+
+
+class FormHeader(BaseModel):
+    title = models.CharField()
+    subtitle = models.CharField(null=True)
+
+
+class SubmissionSettings(BaseModel):
+    role = models.ForeignKey('ConfiguredRole', on_delete=models.CASCADE)
+    message = models.CharField(null=True)
+    redirect_url = models.URLField(null=True)
+    submit_label = models.CharField(null=True)
+
+
+class FormField(BaseModel):
```

Comment:
This model is intended to store information for all kinds of columns(`Scalar`, `Foreign key`, or `Reverse Foreign key`). Let me know if this model is missing any required fields or any fields that can be removed.

Reply:
The model seems to be missing the fields needed to construct `nested_fields`.

`nested_fields` should be present for the `ForeignKeyCreateAllowed` and `ReverseForeignKeyField` types. 

It also misses`create_label` and `allow_create`.

Please also refer the TypeScript definitions `ScalarColumnField`, `ForeignKeyField`, and `ReverseForeignKeyField` on how each field kind differs and the fields required for each of them. All of these inherit the BaseField.